### DISCUSSION TITLE
Ultralytics Question

### DISCUSSION THREAD

**Discussion by Jason1-2** [ultralytics/ultralytics#1818] [Q&A]
I need to add an argument into my python code that can scan only one frame at a time, instead of infinite. Normally, a frame gets displayed on my screen but since the camera is "Printing" infinity frames, I can't reach the next part of my code. I found coding that makes up YOLO models.py, results.py, predictor.py... etc. Does anyone know of an argument that can help me only scan one frame at a time instead of infinity. Thank You!!!

**Comment by dh031200** [comment]
@Jason1-2 
Using the generator as shown below, you can scan frame by frame.
Set `stram=True` for generator

You can find more details on [docs/predict](https://docs.ultralytics.com/modes/predict/)


```python
from ultralytics import YOLO

model = YOLO('yolov8n.pt')
results = model.predict(source=src, verbose=True, stream=True)  # `stram=True ` for generator
for r in results:
    # YOUR CODE
```

**Reply by Jason1-2** [reply]
Thank you for responding. Just to get a better understanding I will show you my python code for my project. 
from ultralytics import YOLO
Import RPi.GPIO as GPIO
from PIL import Image

GPIO.setmode(GPIO.BCM)
GPIO.setup(18, GPIO.OUT)

model=YOLO( ‘best.pt’)

img = model.predict(source=0,show=True)

results = model(img)

if any (obj in results.names for obj in [“Bottle”, “Can”, “Cup”]):
    # Turn on output signal 
    GPIO.output(18, GPIO.HIGH)

else:
    # Turn off output signal 
    GPIO.output(18, GPIO.LOW) 

So you're saying replace my results=model.predict with yours? Also, what does verbose=True mean? Once again thank you!!

**Reply by dh031200** [reply]
@Jason1-2 
The important part is `model.predict()` with `stream=True`.
It gives you the result every frame. 
So your code will be executed every frame. And it also ignores infinite print output.
And I've also included an example conditional statement.
Try it with the code below.
```python
from ultralytics import YOLO
import RPi.GPIO as GPIO
from PIL import Image

GPIO.setmode(GPIO.BCM)
GPIO.setup(18, GPIO.OUT)

model=YOLO('best.pt')

results = model.predict(source=0,show=True, stream=True)

for each_frame_result in results:
    if any(model.predictor.model.names[int(cls)] in ['Bottle', 'Can', 'Cup'] for cls in each_frame.boxes.cls):
        # Turn on output signal
        GPIO.output(18, GPIO.HIGH)
    else:
        # Turn off output signal
        GPIO.output(18, GPIO.LOW)
```

`verbose=True` mean print each detect results like this:
```bash
video 1/1 (1/600) /mnt/e/Dev/ultralytics/vids/sample.mp4: 384x640 2 cars, 1 movable_signage, 1 pole, 1 tree_trunk, 62.4ms
video 1/1 (2/600) /mnt/e/Dev/ultralytics/vids/sample.mp4: 384x640 2 cars, 1 movable_signage, 1 pole, 1 tree_trunk, 26.7ms
video 1/1 (3/600) /mnt/e/Dev/ultralytics/vids/sample.mp4: 384x640 3 cars, 1 fire_hydrant, 1 movable_signage, 1 pole, 1 tree_trunk, 17.9ms
video 1/1 (4/600) /mnt/e/Dev/ultralytics/vids/sample.mp4: 384x640 3 cars, 1 fire_hydrant, 1 movable_signage, 1 pole, 1 tree_trunk, 24.5ms
video 1/1 (5/600) /mnt/e/Dev/ultralytics/vids/sample.mp4: 384x640 3 cars, 1 fire_hydrant, 1 movable_signage, 1 pole, 1 tree_trunk, 24.6ms
```

**Reply by Jason1-2** [reply]
The code you provided worked liked a charm!!! Thank you so much!!!

**Reply by dh031200** [reply]
you're welcome. It was fun talking with you.

**Reply by glenn-jocher** [reply]
@dh031200 i'm glad to hear that the solution worked well for you! If you have any more questions or need further assistance in the future, don't hesitate to reach out. Happy coding! 😊🚀
